### PR TITLE
Fixing some typos in the pyplot API documentation

### PR DESCRIPTION
**Pull request by Naereen**
Hi,
This closes almost all the issues I spotted on the pyplot API doc, see #8074.

Here are the issues this PR fixes:
### stackplot
bug matplotlib doc 7
### hist2d
bug matplotlib doc 6
### arrow
bug matplotlib doc 5

### CODE REVIEW

**Comment by WeatherGod** [comment]
haven't looked through it yet, but I just wanted to make sure that the
pyplot.py file was re-generated using boilerplate.py after installing the
source with the fixes (boilerplate.py imports Axes and inspects that to
generate pyplot.py).

**Comment by Naereen** [comment]
I didn't know that `pyplot.py` should not be manually modified (it's not that clear?)... sorry.

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -1608,21 +1609,23 @@ def semilogx(self, *args, **kwargs):
     # @_preprocess_data() # let 'plot' do the unpacking..
     @docstring.dedent_interpd
     def semilogy(self, *args, **kwargs):
-        r"""Make a plot with log scaling on the `y` axis.
+        r"""
+        Make a plot with log scaling on the *y* axis.
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: This is should be reverted it back to backticks (sorry :( we are trying to make things slightly more consistent across the library).

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -1608,21 +1609,23 @@ def semilogx(self, *args, **kwargs):
...
-        basey : scalar > 1
-            Base of the `y` logarithm.
+        basey : float, optional
+            Base of the *y* logarithm. The scalar should be larger
+            than 1.
 
-        subsy : None or iterable
-            The location of the minor yticks. None defaults to
+        subsy : array_like, optional
+            The location of the minor yticks; *None* defaults to
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: Same.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -5173,8 +5182,8 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
         if im.get_clip_path() is None:
             # image does not already have clipping set, clip to axes patch
             im.set_clip_path(self.patch)
-        #if norm is None and shape is None:
-        #    im.set_clim(vmin, vmax)
+        # if norm is None and shape is None:
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: You can just delete all the commented out code. It should not be there in the first place (but I am happy with the change).

File: lib/matplotlib/stackplot.py [hunk 4]
```diff
@@ -23,32 +23,35 @@ def stackplot(axes, x, *args, **kwargs):
...
 
-    *baseline* : ['zero', 'sym', 'wiggle', 'weighted_wiggle']
-                Method used to calculate the baseline. 'zero' is just a
-                simple stacked plot. 'sym' is symmetric around zero and
-                is sometimes called `ThemeRiver`.  'wiggle' minimizes the
-                sum of the squared slopes. 'weighted_wiggle' does the
-                same but weights to account for size of each layer.
-                It is also called `Streamgraph`-layout. More details
-                can be found at http://leebyron.com/streamgraph/.
+    *baseline* :
```
Comment (NelleV) [inline lib/matplotlib/stackplot.py]: Can you remove the "*"? That should be enough for the formatting to be correct.

File: lib/matplotlib/pyplot.py [hunk 5]
```diff
@@ -2785,7 +2791,8 @@ def contour(*args, **kwargs):
         ret = ax.contour(*args, **kwargs)
     finally:
         ax._hold = washold
-    if ret._A is not None: sci(ret)
+    if ret._A is not None:
```
Comment (tacaswell) [inline lib/matplotlib/pyplot.py]: This needs to be fixed in the `boiler_plate.py` which generates this file (or the fixes will be discarded next time we run it).

**Comment by tacaswell** [comment]
Sorry, you immediately found one of the more interesting bits of matplotlib history :stuck_out_tongue_winking_eye: .  We use a code generation script to write `pyplot.py` instead of a just-in-time function wrapping approach because at the time it was written you could not set the docstring on function objects (py 2.2 or 2.1 iirc).

File: lib/matplotlib/stackplot.py (hunk 4 above)
Comment (Naereen) [inline lib/matplotlib/stackplot.py]: Most likely, the same issues are found elsewhere.

Comment (NelleV) [inline lib/matplotlib/stackplot.py]: Yep… We have this everywhere in our code, but this is a mistake that we keep propagating :(
There should be no markup here :)

**Comment by tacaswell** [comment]
@Naereen Are you still interested in working on this?

**Comment by Naereen** [comment]
Hi @tacaswell. Actually, I'm very surprised to see that it has been fixed...
For example, the docstring & doc for [`semilogy`](http://matplotlib.org/api/pyplot_api.html#matplotlib.pyplot.semilogy) is just unreadable but it has been updated since February.

Honestly I fear that no matter the PR I will send there will always be something not good enough about it, so I prefer to decline, sorry... Good luck.

**Comment by tacaswell** [comment]
@Naereen I am sad you feel that way :(  .  This is a failure on the dev teams side (not yours!).  

Please feel free to reach out to me privately (tcaswell@gmail.com).

**Pull request closed without merging**